app/reader/reader_routes.py

```diff
@@ -8,16 +8,20 @@
 
 @reader_bp.route("/")
 def reader_home():
-    posts_from_db = models.Post.query.order_by(desc(models.Post.post_date)).limit(4)
+    #posts_from_db = models.Post.query.order_by(desc(models.Post.post_date)).limit(4)
+    posts_from_db = models.Article.get_random_articles(4)
```

Comment:
hahahahaha, I like this, I interpret it as getting random articles instead of the latest 4 articles? Am I correct about that?

Reply:
So, this is a good example of that @classmethod thing we were talking about. It's so much cleaner to put a method on the Article class that gets a number of random articles with a parameter for the number.

While I also like the idea of getting random articles more, we could easily add a get_last_articles_by_date method to get them in order. Possibilities are endless and I can see both being useful.

Reply:
Interesting, so the get_random_articles is a @classmethod in the Article object, very cool. Now what was the initial and less clean way of doing it?

Reply:
The initial way was executing a query. The line that was removed was the original way. There are two places that random articles are grabbed so it didn't make sense to keep writing that query. 

I also wasn't a fan of dumping that SQLAlchemy stuff in a route. It just looks more organized to me if that business logic is placed in the models as much as possible. I don't have enough experience to say that we can do the same everywhere but I'd like to try. 

Eventually, I'd like to look into a way to have the models return stuff that is immediately usable by the front end. You can see there's still some logic on some of the routes to break apart the SQLAlchemy query objects into a regular dict that the templates can use. I want to move that into the models as much as possible but I'm not sure what options there are, if any. 

I guess overall, I'm just looking to have the routes contain a minimal amount of code. The heavy lifting can be done in the models. It's definitely something we can discuss though. I think it's largely a stylistic choice.